### PR TITLE
Msi stock ui aggregation

### PR DESCRIPTION
**Pull request by jonfres**
- created a new page for stock administration and all needed repositories, models, resource models needed for stock management
- created API for stock management

### Description
- Issue: https://github.com/magento-engcom/magento2/issues/34

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/Inventory/Controller/Adminhtml/Stock/Edit.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+use Magento\Backend\App\Action\Context;
+use Magento\Backend\Model\View\Result\Page;
+use Magento\Framework\Controller\Result\Redirect;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryApi\Api\StockRepositoryInterface;
+use Magento\InventoryApi\Api\Data\StockInterface;
+
+/**
+ * Class Edit
```
Comment (larsroettig) [inline app/code/Magento/Inventory/Controller/Adminhtml/Stock/Edit.php]: remove class comment

File: app/code/Magento/Inventory/Controller/Adminhtml/Stock/Index.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Controller\Adminhtml\Stock;
+
+use Magento\Backend\App\Action;
+use Magento\Backend\Model\View\Result\Page;
+use Magento\Framework\Controller\ResultFactory;
+
+/**
+ * Class Index
```
Comment (larsroettig) [inline app/code/Magento/Inventory/Controller/Adminhtml/Stock/Index.php]: remove class comment

File: app/code/Magento/Inventory/Controller/Adminhtml/Stock/Save.php [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Registry;
+use Magento\InventoryApi\Api\Data\StockInterface;
+use Magento\InventoryApi\Api\Data\StockInterfaceFactory;
+use Magento\InventoryApi\Api\StockRepositoryInterface;
+
+/**
+ * Class Save
```
Comment (larsroettig) [inline app/code/Magento/Inventory/Controller/Adminhtml/Stock/Save.php]: remove class comment

File: app/code/Magento/Inventory/Controller/Adminhtml/Stock/Save.php [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    ) {
+        parent::__construct($context);
+        $this->stockFactory = $stockFactory;
+        $this->stockRepository = $stockRepository;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->registry = $registry;
+    }
+
+    /**
+     * {@inheritdoc}
```
Comment (larsroettig) [inline app/code/Magento/Inventory/Controller/Adminhtml/Stock/Save.php]: use pls  @inheritdoc without {}

File: app/code/Magento/Inventory/Controller/Adminhtml/Stock/Save.php [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $requestData = $this->getRequest()->getParam('general');
+        if ($this->getRequest()->isPost() && $requestData) {
+            try {
+                $stockId = !empty($requestData[StockInterface::STOCK_ID])
```
Comment (larsroettig) [inline app/code/Magento/Inventory/Controller/Adminhtml/Stock/Save.php]: non-readable negative short if improve this for better readable code

File: app/code/Magento/Inventory/Controller/Adminhtml/Stock/Save.php [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+            }
+        } else {
+            $this->messageManager->addErrorMessage(__('Wrong request.'));
+            $this->processRedirectAfterFailureSave($resultRedirect);
+        }
+        return $resultRedirect;
+    }
+
+    /**
+     * @param int $stockId
```
Comment (larsroettig) [inline app/code/Magento/Inventory/Controller/Adminhtml/Stock/Save.php]: add some  function description

File: app/code/Magento/Inventory/view/adminhtml/ui_component/inventory_stock_listing.xml [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+                <label translate="true">Name</label>
+                <editor>
+                    <editorType>text</editorType>
+                    <validation>
+                        <rule name="required-entry" xsi:type="boolean">true</rule>
+                    </validation>
+                </editor>
+            </settings>
+        </column>
+        <!--<column name="contact_name" sortOrder="40">
```
Comment (larsroettig) [inline app/code/Magento/Inventory/view/adminhtml/ui_component/inventory_stock_listing.xml]: Comment code pls remove

File: app/code/Magento/Inventory/Model/ResourceModel/Stock/Collection.php [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    ) {
+        parent::__construct(
+            $entityFactory,
+            $logger,
+            $fetchStrategy,
+            $eventManager,
+            $connection,
+            $resource
+        );
+    }
```
Comment (maghamed) [inline app/code/Magento/Inventory/Model/ResourceModel/Stock/Collection.php:45]: why do we need constructor, if all the parameters are just proxying to parent constructor

File: app/code/Magento/Inventory/Model/Stock.php [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model;
+
+use Magento\Framework\Model\AbstractExtensibleModel;
+use Magento\InventoryApi\Api\Data\StockInterface;
+
+/**
+ * Class Source
```
Comment (maghamed) [inline app/code/Magento/Inventory/Model/Stock.php:12]: please provide more meaningful description

File: app/code/Magento/Inventory/Model/StockRepository.php [hunk 10]
```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
```
Comment (maghamed) [inline app/code/Magento/Inventory/Model/StockRepository.php:3]: we need to have delete()
operation in Stock repository.

When Stock is removed, we need to remove all stock items related to it (could be done using Foreign Key on DB level)

File: app/code/Magento/InventoryApi/Api/Data/SourceStockLinkInterface.php [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventoryApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+
+/**
+ * SourceCarrierLink interface
```
Comment (maghamed) [inline app/code/Magento/InventoryApi/Api/Data/SourceStockLinkInterface.php:11]: we need to provide more meaningful description to this interface.

File: app/code/Magento/InventoryApi/Api/Data/SourceStockLinkInterface.php [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+ * @api
+ */
+interface SourceStockLinkInterface extends ExtensibleDataInterface
```
Comment (maghamed) [inline app/code/Magento/InventoryApi/Api/Data/SourceStockLinkInterface.php:14]: there is no class which implements this interface

File: app/code/Magento/InventoryApi/Api/Data/SourceStockLinkInterface.php [hunk 13]
```diff
@@ -0,0 +1,68 @@
+<?php
```
Comment (maghamed) [inline app/code/Magento/InventoryApi/Api/Data/SourceStockLinkInterface.php:1]: we need to have dedicated Service interfaces (and implementations) which responsibility is to assign Sources to particular Stock

Something similar to:

Magento\Catalog\Api\CategoryLinkManagementInterface
Magento\Catalog\Api\CategoryLinkRepositoryInterface

File: app/code/Magento/InventoryApi/Api/Data/StockInterface.php [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    public function getName();
+
+    /**
+     * Set source name.
+     *
+     * @param string $name
+     * @return void
+     */
+    public function setName($name);
```
Comment (maghamed) [inline app/code/Magento/InventoryApi/Api/Data/StockInterface.php:50]: need to specify 
getExtensionAttributes/setExtensionAttributes methods and specify own types they accept

File: app/code/Magento/InventoryApi/Api/Data/StockSearchResultsInterface.php [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventoryApi\Api\Data;
+
+use Magento\Framework\Api\SearchResultsInterface;
+
+/**
+ * @api
```
Comment (maghamed) [inline app/code/Magento/InventoryApi/Api/Data/StockSearchResultsInterface.php:11]: please add description to all interfaces marked as api

File: app/code/Magento/InventoryApi/Api/StockRepositoryInterface.php [hunk 16]
```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventoryApi\Api;
+
+/**
+ * @api
```
Comment (maghamed) [inline app/code/Magento/InventoryApi/Api/StockRepositoryInterface.php:9]: meaningful description should be added

File: app/code/Magento/InventoryApi/Api/StockRepositoryInterface.php [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * Load Stock data collection by given search criteria.
+     *
+     * @param \Magento\Framework\Api\SearchCriteriaInterface $searchCriteria
+     * @return \Magento\InventoryApi\Api\Data\StockSearchResultsInterface
+     */
+    public function getList(
+        \Magento\Framework\Api\SearchCriteriaInterface $searchCriteria = null
+    );
```
Comment (maghamed) [inline app/code/Magento/InventoryApi/Api/StockRepositoryInterface.php:40]: need to add deleteById method to the interface

File: app/code/Magento/InventoryApi/Api/StockRepositoryInterface.php [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+     */
+    public function get($stockId);
+
+    /**
+     * Load Stock data collection by given search criteria.
+     *
+     * @param \Magento\Framework\Api\SearchCriteriaInterface $searchCriteria
+     * @return \Magento\InventoryApi\Api\Data\StockSearchResultsInterface
+     */
+    public function getList(
```
Comment (maghamed) [inline app/code/Magento/InventoryApi/Api/StockRepositoryInterface.php:38]: we have the only field (Name) by which we could make filtering, don't think that we need to over complicate our API and accept SearchCriteria in this case.
$name parameter should be okay

**Pull request merged**